tests/test_parser_utils.py

```diff
@@ -442,3 +466,15 @@ def test_v1_2_type_for_source_with_id() -> None:
         cwl_obj, cwl_obj.loadingOptions.fileuri + "#step1/echo_out_file"
     )
     assert source_type == "File"
+
+
+def test_v1_2_type_for_source_with_multiple_entries() -> None:
+    """Test that the type is correctly inferred from a list of source ids with CWL v1.2."""
+    uri = Path(HERE / "../testdata/count-lines7-wf_v12.cwl").resolve().as_uri()
+    cwl_obj = load_document_by_uri(uri)
+    source_type = cwl_utils.parser.cwl_v1_2_utils.type_for_source(
+        cwl_obj, cwl_obj.steps[0].in_[0].source
+    )
+    assert isinstance(source_type, cwl_utils.parser.cwl_v1_2.ArraySchema)
+    assert source_type.items[0].type == "array"
```

Comment:
If here we assert `len(source_type.items) == 1`, we are indirectly testing that comparison between CWL type objects (introduced [here](https://github.com/common-workflow-language/schema_salad/pull/593)) is working correctly. 

However, maybe it is better to open a different PR for that, to avoid superimposing things.

What do you think?